### DISCUSSION TITLE
How to add precheck and pre scripts before git commit

### DISCUSSION THREAD

**Discussion by Nishant-Budia-Jar** [community/community#84930] [Programming Help]
### Select Topic Area

Question

### Body

How to add precheck and pre scripts before git commit

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @Nishant-Budia-Jar ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions.

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by faridnec** [comment]
you can probably try git hooks, these are scripts that Git executes before or after certain events, such as committing, merging, or pushing. You can place your custom scripts in the appropriate hook file, and Git will run them automatically. For adding pre-checks and pre-scripts before a commit, you can use the pre-commit hook.

**Comment by olivierverwoerd** [comment]
Check out [Husky](https://typicode.github.io/husky/)
